### PR TITLE
Implement Burmy form change

### PR DESCRIPTION
**Pull request by BuffelSaft**
## Description
Burmy changes form based on where it last battled. It must actually appear in battle to change form, and will still change form if it fainted in the battle (I've tested this in Platinum and this is the correct behavior).

## **Discord contact info**
Buffel Saft#2205

### CODE REVIEW

File: src/battle_util.c [hunk 1]
```diff
@@ -8659,3 +8659,43 @@ bool32 IsEntrainmentTargetOrSimpleBeamBannedAbility(u16 ability)
     }
     return FALSE;
 }
+
++void DoBurmyFormChange(u32 monId)
```
Comment (LOuroboros) [inline src/battle_util.c:8663]:
```suggestion
void DoBurmyFormChange(u32 monId)
```

Comment (BuffelSaft) [reply src/battle_util.c]: Oops

File: src/battle_util.c [hunk 2]
```diff
@@ -8659,3 +8659,43 @@ bool32 IsEntrainmentTargetOrSimpleBeamBannedAbility(u16 ability)
...
+void DoBurmyFormChange(u32 monId)
+{
+    u16 newSpecies, currSpecies;
+    s32 sentIn;
+    struct Pokemon *party = gPlayerParty;
+
+    sentIn = gSentPokesToOpponent[(gBattlerFainted & 2) >> 1];
+    currSpecies = GetMonData(&party[monId], MON_DATA_SPECIES, NULL);
+
+    if ((GET_BASE_SPECIES_ID(currSpecies) == SPECIES_BURMY) && (gBitTable[monId] & sentIn))
```
Comment (LOuroboros) [inline src/battle_util.c:8672]: `SPECIES_BURMY` is undefined in the battle_engine branch. Either add it to `include/constants/battle_config.h`, or wrap the code with a `#ifdef POKEMON_EXPANSION`.

File: src/battle_util.c [hunk 3]
```diff
@@ -8659,3 +8659,43 @@ bool32 IsEntrainmentTargetOrSimpleBeamBannedAbility(u16 ability)
...
+            case BATTLE_TERRAIN_GRASS:
+            case BATTLE_TERRAIN_LONG_GRASS:
+            case BATTLE_TERRAIN_POND:
+            case BATTLE_TERRAIN_MOUNTAIN:
+            case BATTLE_TERRAIN_PLAIN:
+                newSpecies = SPECIES_BURMY;
+                break;
+            case BATTLE_TERRAIN_CAVE:
+            case BATTLE_TERRAIN_SAND:
+                newSpecies = SPECIES_BURMY_SANDY_CLOAK;
```
Comment (LOuroboros) [inline src/battle_util.c:8685]: `SPECIES_BURMY_SANDY_CLOAK` is undefined in the battle_engine branch. Either add it to `include/constants/battle_config.h`, or wrap the code with a `#ifdef POKEMON_EXPANSION`.

File: src/battle_util.c [hunk 4]
```diff
@@ -8659,3 +8659,43 @@ bool32 IsEntrainmentTargetOrSimpleBeamBannedAbility(u16 ability)
...
+                break;
+            case BATTLE_TERRAIN_BUILDING:
+                newSpecies = SPECIES_BURMY_TRASH_CLOAK;
```
Comment (LOuroboros) [inline src/battle_util.c:8688]: `SPECIES_BURMY_TRASH_CLOAK` is undefined in the battle_engine branch. Either add it to `include/constants/battle_config.h`, or wrap the code with a `#ifdef POKEMON_EXPANSION`.

Comment (BuffelSaft) [reply src/battle_util.c:8688]: I've just added all the Burmy forms to `include/constants/battle_config.h`.

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**